uint8 px
uint8 py
uint8 id 

